delta-test/src/main/java/io/cdap/delta/test/mock/MockErrorTarget.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.delta.test.mock;
+
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.plugin.PluginClass;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.delta.api.Configurer;
+import io.cdap.delta.api.DDLEvent;
+import io.cdap.delta.api.DMLEvent;
+import io.cdap.delta.api.DMLOperation;
+import io.cdap.delta.api.DeltaSource;
+import io.cdap.delta.api.DeltaTarget;
+import io.cdap.delta.api.DeltaTargetContext;
+import io.cdap.delta.api.EventConsumer;
+import io.cdap.delta.api.ReplicationError;
+import io.cdap.delta.api.Sequenced;
+import io.cdap.delta.api.assessment.StandardizedTableDetail;
+import io.cdap.delta.api.assessment.TableAssessment;
+import io.cdap.delta.api.assessment.TableAssessor;
+import io.cdap.delta.proto.Artifact;
+import io.cdap.delta.proto.Plugin;
+
+import java.util.Collections;
+import java.util.HashMap;
+
+/**
+ * Mock target used in unit tests. It only generate error metric.
```

Comment:
nit: "It only set errors to context"

Actually Target just set errors to context and it's context decide whether the emit the error metric immediately once error is set.

Reply:
fixed.